app/config/errors.php

```diff
@@ -10,61 +10,64 @@
+        'description' => 'The request originated from an unknown origin. If you trust this domain, please add it as a platform in the Overview page of your Appwrite Console.',
         'code' => 403,
     ],
     Exception::GENERAL_SERVICE_DISABLED => [
         'name' => Exception::GENERAL_SERVICE_DISABLED,
-        'description' => 'The requested service is disabled. You can enable the service from the Appwrite console.',
+        'description' => 'The requested service is disabled. You can enable the service in the settings of your Appwrite project, under the Services section.',
         'code' => 503,
     ],
     Exception::GENERAL_UNAUTHORIZED_SCOPE => [
         'name' => Exception::GENERAL_UNAUTHORIZED_SCOPE,
-        'description' => 'The current user or API key does not have the required scopes to access the requested resource.',
+        'description' => 'The current user or API key does not have the required scopes to access the requested resource. Update the permissions of the requested resource or add appropriate scopes to your API key.',
         'code' => 401,
     ],
     Exception::GENERAL_RATE_LIMIT_EXCEEDED => [
         'name' => Exception::GENERAL_RATE_LIMIT_EXCEEDED,
-        'description' => 'Rate limit for the current endpoint has been exceeded. Please try again after some time.',
+        'description' => 'Rate limit for the current endpoint has been exceeded. Please try again after some time. Learn more about rate-limits here https://appwrite.io/docs/rate-limits.',
         'code' => 429,
     ],
     Exception::GENERAL_SMTP_DISABLED => [
         'name' => Exception::GENERAL_SMTP_DISABLED,
-        'description' => 'SMTP is disabled on your Appwrite instance. You can <a href="/docs/email-delivery">learn more about setting up SMTP</a> in our docs.',
+        'description' => 'Email delivery is not possible because SMTP is disabled on your Appwrite instance. You can learn more about setting up SMTP here: https://appwrite.io/docs/email-delivery.',
         'code' => 503,
     ],
     Exception::GENERAL_PHONE_DISABLED => [
         'name' => Exception::GENERAL_PHONE_DISABLED,
-        'description' => 'Phone provider is not configured. Please check the _APP_SMS_PROVIDER environment variable of your Appwrite server.',
+        'description' => 'SMS delivery is not possible because a phone provider is not configured. You can learn more about setting up SMS delivery here: https://appwrite.io/docs/sms-delivery.',
         'code' => 503,
     ],
     Exception::GENERAL_ARGUMENT_INVALID => [
         'name' => Exception::GENERAL_ARGUMENT_INVALID,
-        'description' => 'The request contains one or more invalid arguments. Please refer to the endpoint documentation.',
+        // TODO: This is used by the DB, can we make this more specific instead of a generic error or add some kind of templating into the message?
```

Comment:
Argument invalid is very generic and difficult to debug, what would be a better way to handle these errors? I'm unfamiliar with the exact logic behind the many places that use this error.

Reply:
This description is not used/shown to the user when the API throws this error. The message is determined by what the validator found to be incorrect:

https://github.com/appwrite/appwrite/blob/27adf0788ff9809b9e3853480e498e5e169c81b2/app/controllers/api/databases.php#L2907